internal/storage/download.go

```diff
@@ -101,6 +101,15 @@ func downloadArtifact(to string, artifact *Artifact, progress progressBytes, ser
 
 func resume(to string, offset int64, artifact *Artifact, progress progressBytes, serverCert string, retryCount int,
 	retryInterval time.Duration, done chan struct{}) (int64, error) {
+	if offset == int64(artifact.Size) {
+		logger.Infof("validating previously downloaded artifact: %s", to)
+		if err := validate(to, artifact.HashType, artifact.HashValue); err == nil || retryCount == 0 {
```

Comment:
This logic is duplicated with downloadFile method. Is it possible to check the size and offline and if they are equal, downloadFile can be ,called? Of course, the copy there should be skipped.